### PR TITLE
feat: support alter table

### PR DESCRIPTION
**Pull request by tobegit3hub**
* Support add symbolic path.
* Support drop symbolic path.

These are the simple cases to test.

```
create database if not exists db;

create table if not exists db.t1 (name string);

alter table db.t1 add offline_path 'hdfs://foo/bar';

alter table db.t1 drop offline_path 'hdfs://foo/bar';

alter table db.t1 add offline_path 'hdfs://foo/bar';

alter table db.t1 drop offline_path 'hdfs://foo/bar', add offline_path 'hdfs://foo/bar2';

desc db.t1;
```

### CODE REVIEW

File: src/sdk/sql_cluster_router.cc [hunk 1]
```diff
@@ -2643,6 +2643,88 @@ std::shared_ptr<hybridse::sdk::ResultSet> SQLClusterRouter::ExecuteSQL(
...
+                        index += 1;
+                    }
+                    if (isExist) {
+                        offline_table_info.mutable_symbolic_paths()->erase(
+                            offline_table_info.mutable_symbolic_paths()->begin() + index);
+                    } else {
+                        *status = {StatusCode::kCmdError, "The symbolic path does not exist: " + target};
+                        return {};
+                    }
+                    LOG(WARNING) << "Delete path";
```
Comment (tobegit3hub) [inline src/sdk/sql_cluster_router.cc]: Remove the debug code.

File: src/sdk/sql_cluster_test.cc [hunk 2]
```diff
@@ -1050,6 +1050,51 @@ TEST_F(SQLClusterTest, ClusterOnlineAgg) {
...
+    std::string db = "db" + GenRand();
+    ::hybridse::sdk::Status status;
+    bool ok = router->CreateDB(db, &status);
+    ASSERT_TRUE(ok);
+    std::string ddl = "create table (col1 int)";
+    ok = router->ExecuteDDL(db, ddl, &status);
+    ASSERT_TRUE(ok);
+    ASSERT_TRUE(router->RefreshCatalog());
+
+    router->ExecuteSQL(db, "use " + db + ";", &status);
```
Comment (tobegit3hub) [inline src/sdk/sql_cluster_test.cc]: Remove `use`.

File: src/sdk/sql_cluster_test.cc [hunk 3]
```diff
@@ -1050,6 +1050,51 @@ TEST_F(SQLClusterTest, ClusterOnlineAgg) {
     ASSERT_TRUE(ok);
 }
 
+TEST_F(SQLClusterTest, AlterTableAddDropOfflinePath) {
```
Comment (tobegit3hub) [inline src/sdk/sql_cluster_test.cc:1057]: Add the cases for multiple threads.

File: src/sdk/sql_cluster_test.cc [hunk 4]
```diff
@@ -1050,6 +1050,51 @@ TEST_F(SQLClusterTest, ClusterOnlineAgg) {
...
+    ok = router->ExecuteDDL(db, ddl, &status);
+    ASSERT_TRUE(ok);
+    ASSERT_TRUE(router->RefreshCatalog());
+
+    router->ExecuteSQL(db, "use " + db + ";", &status);
+
+    ddl = "ALTER TABLE " + table + " ADD offline_path 'hdfs://foo/bar'";
+    ok = router->ExecuteDDL(db, ddl, &status);
+    ASSERT_TRUE(ok);
+    ASSERT_TRUE(router->RefreshCatalog());
```
Comment (tobegit3hub) [inline src/sdk/sql_cluster_test.cc:1077]: Use tableInfo to check the offline symbolic paths.

**Review by dl239 — APPROVED** [review]

**Review by vagetablechicken — APPROVED** [review]

**Pull request merged**